www/include/configuration/configObject/host/DB-Func.php

```diff
@@ -958,11 +1022,11 @@ function insertHost($ret, $macro_on_demand = null, $server_id = null)
                     $my_tab[$macInput] = str_replace("\$", "", $my_tab[$macInput]);
                     $macName = $my_tab[$macInput];
                     $macVal = $my_tab[$macValue];
-                    $rq = "INSERT INTO on_demand_macro_host (`host_macro_name`, `host_macro_value`,
-                           `description`, `host_host_id`, `macro_order`)
-                           VALUES ('\$_HOST" . strtoupper($macName) . "\$', '" . CentreonDB::escape($macVal) . "', "
-                        . $host_id['MAX(host_id)'] . ", " . $i . ")";
-                    $dbResult = $pearDB->query($rq);
+                    $statement->bindValue(':host_macro_name', '\$_HOST' . strtoupper($macName) . '\$', \PDO::PARAM_STR);
+                    $statement->bindValue(':host_macro_value', $macVal, \PDO::PARAM_STR);
+                    $statement->bindValue(':host_host_id', (int) $host_id['MAX(host_id)'], \PDO::PARAM_INT);
+                    $statement->bindValue(':macro_order', (int) $i, \PDO::PARAM_INT);
```

Comment:
do not always cast please :)
```suggestion
                    $statement->bindValue(':macro_order', $i, \PDO::PARAM_INT);
```

Reply:
@kduret Yes my bad :)